lib/Onyx.js

```diff
@@ -1082,15 +1082,18 @@ function multiSet(data) {
 function applyMerge(existingValue, changes) {
     const lastChange = _.last(changes);
 
-    if (_.isArray(existingValue) || _.isArray(lastChange)) {
+    if (_.isArray(lastChange)) {
         return lastChange;
     }
 
-    if (_.isObject(existingValue) || _.every(changes, _.isObject)) {
-        // Object values are merged one after the other
-        // lodash adds a small overhead so we don't use it here
-        // eslint-disable-next-line prefer-object-spread, rulesdir/prefer-underscore-method
-        return _.reduce(changes, (modifiedData, change) => utils.fastMerge(modifiedData, change),
+    if (_.some(changes, _.isObject)) {
+        // There might be nullish values in a batch of changes that we want to ignore when merging objects,
+        // because otherwise they overwrite the previous changes in the batch.
```

Comment:
Why would we ignore them? Isn't that expected that they would overwrite the previous changes.

Reply:
I think a `null` value would basically cancel the previous ones...

e.g. 

```
Onyx.merge('test', {a: 1});
Onyx.merge('test', null);
Onyx.merge('test', {b: 2});
```
Should give use `{b: 2}`

If we filter out the null values then we are merging the first call with the last and would get `{a: 1, b: 2}`.

Reply:
I think ignoring `undefined` makes sense. But not `null`.

Reply:
> I think ignoring undefined makes sense. But not null.

I think I buy this. Using `null` is a deliberate action to delete something, so it should not be ignored.

Reply:
@marcaaron you're right, this shouldn't filter for null values. Updated 👍